stdlib/testing/testdata/keep_non_existent_only.flux

```diff
@@ -0,0 +1,47 @@
+package testdata_test
+ 
+import "testing"
+
+option now = () => (2030-01-01T00:00:00Z)
+
+inData = "
+#datatype,string,long,dateTime:RFC3339,double,string,string,string,string
+#group,false,false,false,false,true,true,true,true
+#default,_result,,,,,,,
+,result,table,_time,_value,_field,_measurement,old,host
+,,0,2018-05-22T19:53:26Z,0,usage_guest,cpu,cpu-total,host.local
+,,0,2018-05-22T19:53:36Z,0,usage_guest,cpu,cpu-total,host.local
+,,0,2018-05-22T19:53:46Z,0,usage_guest,cpu,cpu-total,host.local
+,,0,2018-05-22T19:53:56Z,0,usage_guest,cpu,cpu-total,host.local
+,,0,2018-05-22T19:54:06Z,0,usage_guest,cpu,cpu-total,host.local
+,,0,2018-05-22T19:54:16Z,0,usage_guest,cpu,cpu-total,host.local
+,,1,2018-05-22T19:53:26Z,0,usage_guest_nice,cpu,cpu-total,host.local
+,,1,2018-05-22T19:53:36Z,0,usage_guest_nice,cpu,cpu-total,host.local
+,,1,2018-05-22T19:53:46Z,0,usage_guest_nice,cpu,cpu-total,host.local
+,,1,2018-05-22T19:53:56Z,0,usage_guest_nice,cpu,cpu-total,host.local
+,,1,2018-05-22T19:54:06Z,0,usage_guest_nice,cpu,cpu-total,host.local
+,,1,2018-05-22T19:54:16Z,0,usage_guest_nice,cpu,cpu-total,host.local
+,,2,2018-05-22T19:53:26Z,91.7364670583823,usage_idle,cpu,cpu-total,host.local
+,,2,2018-05-22T19:53:36Z,89.51118889861233,usage_idle,cpu,cpu-total,host.local
+,,2,2018-05-22T19:53:46Z,91.0977744436109,usage_idle,cpu,cpu-total,host.local
+,,2,2018-05-22T19:53:56Z,91.02836436336374,usage_idle,cpu,cpu-total,host.local
+,,2,2018-05-22T19:54:06Z,68.304576144036,usage_idle,cpu,cpu-total,host.local
+,,2,2018-05-22T19:54:16Z,87.88598574821853,usage_idle,cpu,cpu-total,host.local
+"
+
+outData = "
+#datatype,string,long
+#group,false,false
+#default,_result,
+,result,table
+,,0
```

Comment:
I was surprised by this result.  If we "keep" a non-existent column, I would have expected the same number of rows as in the input tables, all grouped into one table, since there can be no columns in the group key.  In general, I don't expect `keep` to change the number of rows.

Maybe zero rows would also be okay, but a single row seems strange to me.

Reply:
I agree that a single row is incorrect, it should be either the same number of rows as the input table or it should just be an empty table.

An empty table is probably best at this point since none of our code will be able to consume a table with N rows but 0 columns. For example count will not be able to count the rows without at least one column present.

Reply:
my guess is that that "row" is being added by the encoder.  if you look at the code, it's not possible that `drop` or `keep` add rows in this case

if it's the encoder, then I would say that is a separate issue of how to encode a table with no columns.